Makefile

```diff
@@ -1,20 +1,30 @@
-# Minimal makefile for Sphinx documentation
-#
+.PHONY: format lint check test ci help
 
-# You can set these variables from the command line, and also
-# from the environment for the first two.
-SPHINXOPTS    ?=
-SPHINXBUILD   ?= sphinx-build
-SOURCEDIR     = docs
-BUILDDIR      = build
+default: help
 
-# Put it first so that "make" without argument is like "make help".
-help:
-	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+format:
+	@echo "\n\033[1;34m==> Formatting code with Black...\033[0m"
+	black src
+
+lint:
+	@echo "\n\033[1;34m==> Running lint checks with ruff...\033[0m"
+	ruff src
+	@echo "\n\033[1;34m==> Running black check...\033[0m"
+	black --check src
+	@echo "\n\033[1;34m==> Running mypy checks...\033[0m"
+	mypy --strict --disable-error-code=type-arg --disable-error-code=no-untyped-call --ignore-missing-import --disable-error-code=unused-ignore src
 
-.PHONY: help Makefile
+test:
+	@echo "\n\033[1;34m==> Running tests...\033[0m"
+	pytest
 
-# Catch-all target: route all unknown targets to Sphinx using the new
-# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
-%: Makefile
```

Comment:
sad that the sphinx docs are broken now. We'll eventually have to redo these docs.